web/src/components/studies/CreateStudyForm.tsx

```diff
@@ -0,0 +1,371 @@
+  ssr: false,
+});
+const AlertMessage = dynamic(() => import("uikit-react-public").then((mod) => mod.Alert.Message), {
+  ssr: false,
+});
+
+type CreateStudyProps = {
+  username: string;
+  setCreateStudyFormOpen: (name: boolean) => void;
+};
+
+interface CreateStudyValues {
+  shortStudyName: string;
+  longStudyName: string;
+  owner: string;
+  admin: string;
+  controller: string;
+  cagRef: number;
+  dataProtectionNumber: number;
+  nhsEnglandRef: number;
+  irasId: string; // might be number, unclear
+  // checkboxes
+  uclSponsorship: boolean;
+  cag: boolean;
+  ethics: boolean;
+  hra: boolean;
+  dbs: boolean;
+  dataProtection: boolean;
+  thirdParty: boolean;
+  externalUsers: boolean;
+  consent: boolean;
+  nonConsent: boolean;
+  extEea: boolean;
+  nhs: boolean;
+  nhsEngland: boolean;
+  mnca: boolean;
+  dspt: boolean;
+}
+
+export default function CreateStudyForm(CreateStudyProps: CreateStudyProps) {
+  const { username, setCreateStudyFormOpen } = CreateStudyProps;
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors, isValid, isSubmitting },
+  } = useForm<CreateStudyValues>({
+    mode: "onChange",
+    criteriaMode: "all",
+    defaultValues: {
+      owner: username,
+    },
+  });
+  const [currentStep, setCurrentStep] = useState(1);
+  const nextStep = () => setCurrentStep(currentStep + 1);
+  const prevStep = () => setCurrentStep(currentStep - 1);
+  const totalSteps = 3;
+
+  const showCagRef = useWatch({
+    name: "cag",
+    control,
+  });
+  const showIrasId = useWatch({
+    name: "hra",
+    control,
+  });
+  const showDataProtectionNumber = useWatch({
+    name: "dataProtection",
+    control,
+  });
+  const showNhsRelated = useWatch({
+    name: "nhs",
+    control,
+  });
+  const showNhsEnglandRef = useWatch({
+    name: "nhsEngland",
+    control,
+  });
+
+  const onSubmit: SubmitHandler<CreateStudyValues> = (data) => {
+    console.log(data);
+    //todo: do the things
+  };
+  const getFieldsetClass = (step: number) =>
+    `${styles.fieldset} ${currentStep === step ? styles.visible : styles.hidden}`;
+
+  return (
+    <Dialog setDialogOpen={setCreateStudyFormOpen} className={styles["study-dialog"]} cy="create-study-form">
+      <h2>Create Study</h2>
+      <div className={styles["step-progress"]}>
+        <div
+          className={`${styles["step-dot"]} ${currentStep === 1 ? styles["active"] : ""} ${isValid ? styles.valid : ""}`}
+        ></div>
+        <div className={`${styles["step-dot"]} ${currentStep === 2 ? styles["active"] : ""}`}></div>
+        <div className={`${styles["step-dot"]} ${currentStep === 3 ? styles["active"] : ""}`}></div>
+      </div>
+
+      <form className={styles.form} onSubmit={handleSubmit(onSubmit)}>
+        {/* first step */}
+        <fieldset className={getFieldsetClass(1)}>
+          <legend>Study Name</legend>
+          <Label htmlFor="shortStudyName">
+            Short Study Name*:
```

Comment:
personal preference:
- "Short Study Name" -> "Study Name"
- "Long Study Name" -> "Study Description"

Reply:
should we enforce uniqueness of the study name – I think probably

Reply:
@finleybacon any reason we shouldn't do this? If not then I'm on board

Reply:
for the uniqueness, would the best way be to send a request to the db to check if there's already a study with that name or is there another way?